src/collectors/ceph/ceph.py

```diff
@@ -196,9 +196,15 @@ def _parse_socket_name(self, path):
         """Parse a socket name like /var/run/ceph/foo-osd.2.asok
 
         Return a 3 tuple of cluster name, service type, service id
+        or None if we don't parse it correctly
         """
-        return re.match("^(.*)-(.*)\.(.*).{0}$".format(self.config['socket_ext']),
-                        os.path.basename(path)).groups()
+        try:
+            cluster_name, service_type, service_id = re.match("^(.*)-(.*)\.(.*).{0}$".format(self.config['socket_ext']),
+                                                              os.path.basename(path)).groups()
+        except AttributeError:
+            return None
```

Comment:
Looking below, you're having to do an explicit "if ... is None" on each call to this function, so it would not be any more painful to use an exception handler around each call instead.